### PR TITLE
Add managed resources to the operator status

### PR DESCRIPTION
**Pull request by skitt**
This adds the various daemon sets to the operator status: the engine
daemonset, the route agent daemonset, and the globalnet daemonset.

Currently the status of these daemon sets isn’t updated outside the
reconciliation loop, which means the status is inaccurate.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

File: deploy/crds/submariner.io_submariners_crd.yaml [hunk 1]
```diff
@@ -97,10 +97,14182 @@ spec:
               type: string
             colorCodes:
               type: string
+            engineDaemonSet:
```
Comment (dfarrell07) [inline deploy/crds/submariner.io_submariners_crd.yaml:100]: We really need this 14k LoC?

Comment (skitt) [reply deploy/crds/submariner.io_submariners_crd.yaml:100]: Indeed, we don’t, I was thinking of only including the daemonset status instead.

**Comment by skitt** [comment]
In addition to Daniel’s comment, this needs changes to register the daemonsets, otherwise their status is never updated.

File: deploy/crds/submariner.io_submariners_crd.yaml (hunk 1 above)
Comment (tpantelis) [reply deploy/crds/submariner.io_submariners_crd.yaml:100]: This is generated code - the result of completely embedding a Pod object.  I was a little wary of this too  Perhaps we should embed just the Pod's Status?

Comment (tpantelis) [reply deploy/crds/submariner.io_submariners_crd.yaml:100]: Sorry my comment crossed with @skitt comment 😄

**Pull request merged**